101-hdinsight-spark-cmk/azuredeploy.json

```diff
@@ -0,0 +1,193 @@
+                        "isDefault":true,
+                        "container":"[parameters('newOrExistingClusterStorageContainerName')]",
+                        "key":"[listKeys(resourceId(parameters('existingClusterStorageResourceGroup'), 'Microsoft.Storage/storageAccounts', parameters('existingClusterStorageAccountName')), '2018-02-01').keys[0].value]"
+                    }
+                    ]
+                },
+                "computeProfile": {
+                    "roles": [
+                        {
+                            "autoscale": null,
+                            "name": "headnode",
+                            "minInstanceCount": 1,
+                            "targetInstanceCount": 2,
+                            "hardwareProfile": {
+                                "vmSize": "Standard_D12_V2"
+                            },
+                            "osProfile": {
+                                "linuxOperatingSystemProfile": {
+                                    "username": "[parameters('sshUserName')]",
+                                    "password": "[parameters('sshPassword')]"
+                                },
+                                "windowsOperatingSystemProfile": null
+                            },
+                            "virtualNetworkProfile": null,
+                            "scriptActions": [],
+                            "dataDisksGroups": null
+                        },
+                        {
+                            "autoscale": null,
+                            "name": "workernode",
+                            "targetInstanceCount": "[parameters('clusterWorkerNodeCount')]",
+                            "hardwareProfile": {
+                                "vmSize": "Standard_D13_V2"
+                            },
+                            "osProfile": {
+                                "linuxOperatingSystemProfile": {
+                                    "username": "[parameters('sshUserName')]",
+                                    "password": "[parameters('sshPassword')]"
+                                },
+                                "windowsOperatingSystemProfile": null
+                            },
+                            "virtualNetworkProfile": null,
+                            "scriptActions": [],
+                            "dataDisksGroups": null
+                        }
+                    ]
+                },
+                "minSupportedTlsVersion": "[parameters('minTlsVersionNumber')]",
+                "diskEncryptionProperties": {
+                    "vaultUri": "[parameters('diskEncryptionVaultUri')]",
+                    "keyName": "[parameters('diskEncryptionKeyName')]",
+                    "keyVersion": "[parameters('diskEncryptionKeyVersion')]",
+                    "msiResourceId": "[parameters('diskEncryptionMsiResourceId')]"
+                }
+            },
+            "identity": {
```

Comment:
This "identity" would not work here. It needs to be placed as a sibling of "properties".

Reply:
It is under properties

Reply:
It should be a "sibling", not "under". And please make sure you try and this works.

Reply:
What error did you receive when you executed this?